### PR TITLE
Step-3 possible extensions to label boundaries using x, y, z coordinates

### PR DESCRIPTION
**Pull request by maemar**
Current extension uses face labels to change boundary. I added a snippet that labels boundaries using x, y, z coordinates.

### CODE REVIEW

File: examples/step-3/doc/results.dox [hunk 1]
```diff
@@ -124,6 +124,31 @@ suggestions:
   potentially non-zero Neumann conditions). You will see this if you
   run the program.
 
+  An alternative way to change the boundary indicator is to label
+  the boundaries by the cartesian coordinates of the cell centers.
```
Comment (masterleinad) [inline examples/step-3/doc/results.dox]:
```suggestion
  the boundaries by the cartesian coordinates of the face centers.
```

File: examples/step-3/doc/results.dox [hunk 2]
```diff
@@ -124,6 +124,31 @@ suggestions:
...
+  For example, we can label all of the cells along the top and
+  bottom boundaries with a boundary indicator 1 by checking to
+  see if the cell centers' y-coordiantes are within a tolerance
```
Comment (masterleinad) [inline examples/step-3/doc/results.dox]:
```suggestion
  see if the cell centers' y-coordinates are within a tolerance
```

File: examples/step-3/doc/results.dox [hunk 3]
```diff
@@ -124,6 +124,31 @@ suggestions:
...
+  (here 1e-12) of -1 and 1. Try this immediately after calling
+  <code>GridGenerator::hyper_cube</code>, as before:
+  @code
+  for (; cell != endc; ++cell)
+    {
+      {
```
Comment (masterleinad) [inline examples/step-3/doc/results.dox]: There is no need for extra braces. :slightly_smiling_face: 
```suggestion
      {
```

File: examples/step-3/doc/results.dox [hunk 4]
```diff
@@ -124,6 +124,31 @@ suggestions:
...
+        for (unsigned int face_number = 0;
+             face_number < GeometryInfo<2>::faces_per_cell;
+             ++face_number)
+          if ((std::fabs(cell->face(face_number)->center()(1) - (-1)) <
+               1e-12) ||
+              (std::fabs(cell->face(face_number)->center()(1) - (1)) < 1e-12))
+            cell->face(face_number)->set_boundary_id(1);
+      }
```
Comment (masterleinad) [inline examples/step-3/doc/results.dox]:
```suggestion
```

File: examples/step-3/doc/results.dox [hunk 5]
```diff
@@ -124,6 +124,31 @@ suggestions:
...
 
+  An alternative way to change the boundary indicator is to label
+  the boundaries by the cartesian coordinates of the cell centers.
+  For example, we can label all of the cells along the top and
+  bottom boundaries with a boundary indicator 1 by checking to
+  see if the cell centers' y-coordiantes are within a tolerance
+  (here 1e-12) of -1 and 1. Try this immediately after calling
+  <code>GridGenerator::hyper_cube</code>, as before:
+  @code
+  for (; cell != endc; ++cell)
```
Comment (masterleinad) [inline examples/step-3/doc/results.dox]: Can't you use a range-based for loop?

Comment (maemar) [reply examples/step-3/doc/results.dox]: What do you mean by range-based for loop?

Comment (masterleinad) [reply examples/step-3/doc/results.dox]: If you have a look at `Step3::assemble_system`, you will see that the loop over the cells looks like:
```
for (const auto &cell : dof_handler.active_cell_iterators())
{
  ...
}
```
We generally prefer this  syntax over
```
for (; cell != endc; ++cell)
{
  ...
}
```

Comment (maemar) [reply examples/step-3/doc/results.dox]: I see, thanks for the suggestion! Just made the change.

**Pull request merged**